gym/wrappers/flatten_observation.py

```diff
@@ -0,0 +1,8 @@
+from gym.spaces import flatten
+from gym import ObservationWrapper
+
+
+class FlattenObservation(ObservationWrapper):
```

Comment:
If we wrap an environment with this wrapper, then `self.observation_space.shape` won't match the shape of observations returned by the env. I think it should also modify the observation space.